<hr style="height:2px;border:none"/>
<h1 align='center'> INF-280 Estadística Computacional II-2022 </h1>

<H3 align='center'> Tema 8 - Test de Hipótesis  </H3>

\


Nombre del equipo: 

Integrantes:


*   Martin Sarubbi - 201973587-K - [video](https://youtu.be/EK3z4FocSss)
*   Benjamín Vallejos - 201910516-7 - [video](https://www.youtube.com/watch?v=SeNL2fmQm8w)
*   Miguel Soto Delgado - 201973623-K - [video](https://drive.google.com/drive/folders/17FG7pWxCkkUPgBBfAmCoJMoXVYcjhqIp?usp=sharing)


*Recuerden que si su equipo se llama "El ritmo no perdona", entonces el nombre del archivo a subir debe ser T8_El_ritmo_no_perdona.ipynb*


### **Formalidades**  
* Es posible utilizar apuntes, libros, papers, ejemplos y cualquier otro material que desee de internet, pero debe ser usted quien desarrolle los ejercicios y comente los resultados en el Notebook que entregará. 
* Se permite el uso de las librerías: pandas, numpy, matplotlib y seaborn. Si desea utilizar otra librería debe indicar cómo instalarla.
* Se debe trabajar en grupos de **dos o tres personas**, las cuales deberán estar inscritas a través de __[Aula](https://aula.usm.cl/)__.
* Además del Notebook, deberán entregar un video explicando lo que hizo, paso a paso, como en los ejemplos entregados en AULA. Este vídeo es **individual**, es decir, cada integrante debe hacerlo por separado (debiendo utilizar el mismo Jupyter Notebook). El video puede ser subido a una plataforma externa como Youtube, Google Drive, ó Dropbox. El **link a compartir** (*visible para cualquier usuario con el link*) debe ir dentro del Notebook.
* La nota de esta actividad es grupal.


<hr style="height:2px;border:none"/>

### **Entrega** 
* **Formato**: Se debe realizar una **única** entrega por grupo (un representate del grupo) a través de  __[**Aula**](https://aula.usm.cl/)__.
* **Archivo**: El nombre del archivo a subir debe ser **T8_Nombre_Del_Equipo.ipynb**, por ejemplo, si su equipo se llama "El ritmo no perdona", entonces el nombre del archivo debe ser **T8_El_ritmo_no_perdona.ipynb**
* **Archivo**: Sus respuestas deben ser entregadas en forma de Jupyter Notebook, incluyendo **todos links** a los vídeos explicativos, uno por cada integrante.
    * **Vídeo**: Debe grabar la pantalla (Jupyter Notebok) donde se realiza la explicación de la actividad, con una duración máxima de 25 minutos.
* **Fecha límite**: Martes 6 de Diciembre a las 23:59

<hr style="height:2px;border:none"/>

### **Descuentos**

* Se descontarán 5 puntos si el nombre del archivo subido no sigue el formato señalado.
* Se descontarán 10 puntos por cada hora de atraso en la entrega.
* Todos los gráficos que realicen deben contener los nombres de las variables en los ejes. Se descontarán 3 puntos por cada nombre faltante.

<hr style="height:2px;border:none"/>

<table align='left'>
<tr>
<td><img src='https://comunicacion.uaa.mx/revista/wp-content/uploads/2019/01/discurso-620x350.png' width='650'/> </td>
</tr>
</table>

## Contexto

La Universidad de Harvard ha recopilado una serie de discursos para medir el nivel de populismo en los discursos de 241 jefes ejecutivos (presidentes y primeros ministros) de 74 países de todos los continentes. El conjunto de datos cubre 324 mandatos gubernamentales e incluye 1240 discursos, en su mayoría entre 2000 y 2022. 

El dataset `gpd_v2_20220427.csv` está compuesto por los siguientes atriibutos:

1.   `merging_variable`: name of the .txt file in the folder with Speeches with the speech that matches that coder’s score.
2.   `Country`
3.   `Leader`: name of president or prime minister.
4.   `lr`: categorical indicator with -1 = left; 0 = center; 1 = right.
5.   `president`: binary. 1 if the leader is a president, 0 if a prime minister.
6.   `term`: leader term in office, ordinal.
8.   `startofterm`: date of start of respective term
9.   `yearbegin`: year of beginning of term.
10.   `endofterm`: date of end of respective term.
11.   `yearend`: year of ending of respective term.
12.   `speechtype`: type of speech: international, campaign, ribbon-cutting, famous.
13.   `speechnum`: number of that speech for that leader-term.
14.   `codernum`: coder identifier for that speech-leader.
15.   `rubricgrade`: grade of that speech by that coder on the 0-2 populism scale.
16.   `averagerubric`: average populism grade for that speech. Arithmetic average across all coders for that speech.
17.    `totalaverage`: average populism grade for that leader-term. Arithmetic mean for all speeches by all coders for that leader-term
18.    `wb_region`: World Bank global region classification.

# Librerías

In [ ]:
import pandas as pd
import numpy as np 
import seaborn as sns; sns.set()
import matplotlib.pyplot as plt
import warnings
warnings.filterwarnings("ignore")


# Datos

In [ ]:
data = pd.read_csv("gpd_v2_20220427.csv")
data


merging_variable    country          leader  \
0             Albania_Berisha_Campaign_2.txt    Albania    Sali Berisha   
1             Albania_Berisha_Campaign_2.txt    Albania    Sali Berisha   
2             Albania_Berisha_Campaign_2.txt    Albania    Sali Berisha   
3             Albania_Berisha_Campaign_2.txt    Albania    Sali Berisha   
4               Albania_Berisha_Ribbon_2.txt    Albania    Sali Berisha   
...                                      ...        ...             ...   
4955  Australia_Morrison_International_1.txt  Australia  Scott Morrison   
4956         Australia_Morrison_Ribbon_1.txt  Australia  Scott Morrison   
4957         Australia_Morrison_Ribbon_1.txt  Australia  Scott Morrison   
4958         Australia_Morrison_Ribbon_1.txt  Australia  Scott Morrison   
4959         Australia_Morrison_Ribbon_1.txt  Australia  Scott Morrison   

                            party   lr  president  term startofterm  \
0     Democratic Party of Albania  0.0          0     2  2009-06-28   
1     Democratic Party of Albania  0.0          0     2  2009-06-28   
2     Democratic Party of Albania  0.0          0     2  2009-06-28   
3     Democratic Party of Albania  0.0          0     2  2009-06-28   
4     Democratic Party of Albania  0.0          0     2  2009-06-28   
...                           ...  ...        ...   ...         ...   
4955   Liberal Party of Australia  NaN          0     1  2018-04-24   
4956   Liberal Party of Australia  NaN          0     1  2018-04-24   
4957   Liberal Party of Australia  NaN          0     1  2018-04-24   
4958   Liberal Party of Australia  NaN          0     1  2018-04-24   
4959   Liberal Party of Australia  NaN          0     1  2018-04-24   

      yearbegin   endofterm  yearend      speechtype  speechnum  codernum  \
0          2009  2013-06-23     2013        Campaign          1         1   
1          2009  2013-06-23     2013        Campaign          1         2   
2          2009  2013-06-23     2013        Campaign          1         3   
3          2009  2013-06-23     2013        Campaign          1         4   
4          2009  2013-06-23     2013  Ribbon cutting          2         1   
...         ...         ...      ...             ...        ...       ...   
4955       2018     present  present   International          3         4   
4956       2018     present  present  Ribbon cutting          4         1   
4957       2018     present  present  Ribbon cutting          4         2   
4958       2018     present  present  Ribbon cutting          4         3   
4959       2018     present  present  Ribbon cutting          4         4   

      rubricgrade  averagerubric  totalaverage              wb_region  \
0             0.0            0.5       0.12500  Europe & Central Asia   
1             1.0            0.5       0.12500  Europe & Central Asia   
2             NaN            0.5       0.12500  Europe & Central Asia   
3             NaN            0.5       0.12500  Europe & Central Asia   
4             0.0            0.0       0.12500  Europe & Central Asia   
...           ...            ...           ...                    ...   
4955          NaN            0.0       0.15625    East Asia & Pacific   
4956          0.0            0.0       0.15625    East Asia & Pacific   
4957          0.0            0.0       0.15625    East Asia & Pacific   
4958          NaN            0.0       0.15625    East Asia & Pacific   
4959          NaN            0.0       0.15625    East Asia & Pacific   

                          region  
0     Central and Eastern Europe  
1     Central and Eastern Europe  
2     Central and Eastern Europe  
3     Central and Eastern Europe  
4     Central and Eastern Europe  
...                          ...  
4955                         NaN  
4956                         NaN  
4957                         NaN  
4958                         NaN  
4959                         NaN  

[4960 rows x 19 columns]

#Funciones


In [ ]:
#H0: mu1 = mu2 (mu1,mu2 ambas no conocidas)
def contrate_2medias(n1,n2,media1,media2,std1,std2,error_tipo1,alternativa='1>2'):
  region_critica = []
  gl_num = (std1**2/float(n1) + std2**2/float(n2))**2
  gl_den = ((std1**2/float(n1))**2)/(n1-1.0) + ((std2**2/float(n2))**2)/(n2-1.0)
  gl = int(np.floor(gl_num/gl_den))
  std_error = np.sqrt(std1**2/n1 + std2**2/n2)
  T0 = (media1-media2)/std_error#estadistico bajo H0
  if (alternativa=='1>2'):
    region_critica.append(pd.Interval(left=t.ppf(1-error_tipo1,df=gl), right=np.inf,closed='neither'))
    p_value = 1-t.cdf(T0,df=gl)
  elif (alternativa=='1<2'):
    region_critica.append(pd.Interval(left=-np.inf, right=t.ppf(error_tipo1,df=gl),closed='neither'))
    p_value = t.cdf(T0,df=gl)
  else:#dos colas
    region_critica.append(pd.Interval(left=-np.inf,right=t.ppf(error_tipo1/2,df=gl),closed='neither'))
    region_critica.append(pd.Interval(left=t.ppf(1-error_tipo1/2,df=gl),right=np.inf,closed='neither'))
    p_value = 2*min(1-t.cdf(T0,df=gl),t.cdf(T0,df=gl))
  return T0, region_critica, p_value


def contrate_1media(n1,media1,std1,mu0,error_tipo1,alternativa='1>0'):
  region_critica = []
  gl = n1-1
  std_error = np.sqrt(std1**2/n1)
  T0 = (media1-mu0)/std_error#estadistico bajo H0
  if (alternativa=='1>0'):
    region_critica.append(pd.Interval(left=t.ppf(1-error_tipo1,df=gl), right=np.inf,closed='neither'))
    p_value = 1-t.cdf(T0,df=gl)
  elif (alternativa=='1<0'):
    region_critica.append(pd.Interval(left=-np.inf, right=t.ppf(error_tipo1,df=gl),closed='neither'))
    p_value = t.cdf(T0,df=gl)
  else:#dos colas
    region_critica.append(pd.Interval(left=-np.inf,right=t.ppf(error_tipo1/2,df=gl),closed='neither'))
    region_critica.append(pd.Interval(left=t.ppf(1-error_tipo1/2,df=gl),right=np.inf,closed='neither'))
    p_value = 2*min(1-t.cdf(T0,df=gl),t.cdf(T0,df=gl))
  return T0, region_critica, p_value



# Preguntas

---

**Importante** para las siguientes preguntas:
* Indique explicitamente **los supuestos** que consideraron.
* Indique explicitamente **$H_0$** y **$H_1$**.
* Indique explicitamente el valor del **Estadístico de Contraste**.
* Indique explicitamente la **Región Crítica**.
* Indique explicitamente si **se acepta $H_0$** o **se rechaza $H_0$**.

---

1. Considerando solo los discursos realizados en  `region=Latin America & Caribbean` y `region=Western Europe` desde el año 2000 en adelante y solamente aquellos realizados en campaña (`speechtype = Campaign`). 

> a) Diseñe un contraste que permita evaluar la hipótesis de que "*los discursos realizados por presidentes de izquierda (`lr = -1`) tienen un populismo (`rubricgrade`) promedio **distinto** en ambas regiones*". Diseñe su contraste de manera que la probabilidad de aceptar incorrectamente esta conjetura sea menor al 5%. **(10 pts)**

> b) Diseñe un contraste que permita evaluar la hipótesis de que "*los discursos realizados por presidentes de centro (`lr = 0`) tienen un populismo (`rubricgrade`) promedio **distinto** en ambas regiones*". Diseñe un contraste de manera que la probabilidad de aceptar incorrectamente esta conjetura sea menor al 5%. **(10 pts)**

> c) Diseñe un contraste que permita evaluar la hipótesis de que "*los discursos realizados por presidentes de derecha (`lr = 1`) tienen un populismo (`rubricgrade`) promedio **distinto** en ambas regiones*". Diseñe un contraste de manera que la probabilidad de aceptar incorrectamente esta conjetura sea menor al 5%. **(10 pts)**

2. Considerando solo los discursos realizados en Chile desde el año 2000 en adelante. Diseñe un contraste que permita evaluar la hipótesis de que "*los discursos realizados por presidentes de izquierda (`lr = -1`) son **más populistas** (utilice `rubricgrade`) que los discursos realizados por presidentes de derecha (`lr = -1`)*". Diseñe un contraste de manera que la probabilidad de aceptar incorrectamente esta conjetura sea menor al 5%.  **(10 pts)**

3. En Chile, Michelle Bachelet y Sebastián Piñera han sido dos veces presidente de la república. Por esto, podría resultar interesante determinar si sus niveles de populismo (utilice `rubricgrade`) en sus discursos cambiaron en algo durante sus dos mandatos. 

> a) Ejecute un contraste donde $H_0$ corresponda a que "*los discursos de "Bachelet 1" son **igual o más populistas** que los discursos de "Bachelet 2"* ". Considere $α =  5\%$. **(15 pts)**

> b) Ejecute un contraste donde $H_0$ corresponda a que "*los discursos de "Piñera 1" son **igual o más populistas** que los discursos de "Piñera 2"* ". Considere $α =  5\%$. **(15 pts)**

4. En los últimos 20 años Argentina ha aumentado su nivel de pobreza, donde más del 25% de la población del país se encuentra en esta situación.

> a) Diseñe un contraste que permita evaluar la hipótesis de que "*el populismo (utilice `rubricgrade`) promedio en los discursos de presidentes argentinos es mayor a 0.5*". Diseñe un contraste de manera que la probabilidad de aceptar incorrectamente esta conjetura sea menor al 5%. **(10 pts)**

> b) Diseñe un contraste que permita evaluar la hipótesis de que "*los discursos realizados por presidentes argentinos son más populistas que los discursos realizados por el resto de países de `region=Latin America & Caribbean`*". Diseñe un contraste de manera que la probabilidad de aceptar incorrectamente esta conjetura sea menor al 5%  **(10 pts)**

**Debido a los problemas con el dataset de la pregunta 5), se ha decidido eliminar dicha pregunta. Por esto, se otorgará un puntaje base de 20 pts.**

# Ejemplo Test de Hipótesis:


En un test de hipótesis, la hipótesis nula ($H_0$) corresponde al *status quo*, es decir, a lo que se "cree que es lo normal". En cambio, la hipótesis alternativa es lo que se "sospecha" que ocurre.





Ejemplo:

Diseñe un contraste que permita evaluar la hipótesis de que "la media de la población X es distinta a la media de la población Y". Diseñe un contraste de manera que la probabilidad de aceptar incorrectamente esta conjetura sea menor al 5%.

En este caso, se tiene que:
* $H_0$: $μ_X = \mu_Y$ (Corresponde a lo que se cree "normal")
* $H_1$: $μ_X \neq \mu_Y$

Además, se plantean de esta manera las hipótesis ya que "aceptar incorrectamente la conjetura" corresponde a un error tipo 1 (valor de $α$) y se sabe que siempre se debe acotar este tipo de error, pues es más grave que un error tipo 2.

# Respuestas

#Pregunta 1

Considerando solo los discursos realizados en  `region=Latin America & Caribbean` y `region=Western Europe` desde el año 2000 en adelante y solamente aquellos realizados en campaña (`speechtype = Campaign`). 

> a) Diseñe un contraste que permita evaluar la hipótesis de que "*los discursos realizados por presidentes de izquierda (`lr = -1`) tienen un populismo (`rubricgrade`) promedio **distinto** en ambas regiones*". Diseñe su contraste de manera que la probabilidad de aceptar incorrectamente esta conjetura sea menor al 5%. **(10 pts)**

> b) Diseñe un contraste que permita evaluar la hipótesis de que "*los discursos realizados por presidentes de centro (`lr = 0`) tienen un populismo (`rubricgrade`) promedio **distinto** en ambas regiones*". Diseñe un contraste de manera que la probabilidad de aceptar incorrectamente esta conjetura sea menor al 5%. **(10 pts)**

> c) Diseñe un contraste que permita evaluar la hipótesis de que "*los discursos realizados por presidentes de derecha (`lr = 1`) tienen un populismo (`rubricgrade`) promedio **distinto** en ambas regiones*". Diseñe un contraste de manera que la probabilidad de aceptar incorrectamente esta conjetura sea menor al 5%. **(10 pts)**

Para empezar iniciemos creando la hipotesis, la cual es



$H_0$: $μ_L$ $=$  $μ_E$

$H_1$: $μ_L$ $\not =$ $μ_E$

Estas hipotesis son iguales para las tres casos, solo que cambia que los presidentes son de izquieda, centro o derecha

In [ ]:
from scipy.stats import t  
#Parte a

muestra = data[(data['region'] == 'Latin America & Caribbean')| (data['region'] == 'Western Europe')]
muestra = muestra[muestra['speechtype'] =='Campaign']

Izq_Lat = muestra[(muestra['region'] == 'Latin America & Caribbean')| (muestra['lr'] == -1)]
Izq_Lat = Izq_Lat[Izq_Lat['rubricgrade'].notnull()]
Izq_Lat = Izq_Lat['rubricgrade']

Izq_Eu = muestra[(muestra['region'] == 'Western Europe') | (data['lr'] == -1)]
Izq_Eu = Izq_Eu[Izq_Eu['rubricgrade'].notnull()]
Izq_Eu = Izq_Eu['rubricgrade']
#Parametros Latino
N_Izq_Lat = len(Izq_Lat)
media_Izq_Lat = np.mean(Izq_Lat)
std_Izq_Lat = np.std(Izq_Lat)
#Parametros Europa
N_Izq_Eu = len(Izq_Eu)
media_Izq_Eu = np.mean(Izq_Eu)
std_Izq_Eu = np.std(Izq_Eu)


Contraste_Izq = contrate_2medias(N_Izq_Lat,N_Izq_Eu,media_Izq_Lat,media_Izq_Eu,std_Izq_Lat,std_Izq_Eu,0.05,alternativa='1=2')

print("\nComo resultado se obtienen los siguientes valores:\n")
print("     Z0                                                         región crítica                                                           p-value\n")
print(Contraste_Izq)


Como resultado se obtienen los siguientes valores:

     Z0                                                         región crítica                                                           p-value

(0.6357399377546483, [Interval(-inf, -1.9670282846690177, closed='neither'), Interval(1.9670282846690172, inf, closed='neither')], 0.5253774161703222)


a)

Como se puede observar, no se tiene la evidencia suficiente para poder rechazar $H_0$, esto se debe a que el estadistico no esta dentro de la región crítica, por lo tanto se puede tomar en cuenta que los discursos realizados por presidentes de izquierda tienen un populismo similar en las dos regiones. 

In [ ]:
#Parte B
Cen_Lat = muestra[(muestra['region'] == 'Latin America & Caribbean')| (muestra['lr'] == 0)]
Cen_Lat = Cen_Lat[Cen_Lat['rubricgrade'].notnull()]
Cen_Lat = Cen_Lat['rubricgrade']


Cen_Eu = muestra[(muestra['region'] == 'Western Europe') | (data['lr'] == 0)]
Cen_Eu = Cen_Eu[Cen_Eu['rubricgrade'].notnull()]
Cen_Eu = Cen_Eu['rubricgrade']
#Parametros Latino
N_Cen_Lat = len(Cen_Lat)
media_Cen_Lat = np.mean(Cen_Lat)
std_Cen_Lat = np.std(Cen_Lat)
#Parametros Europa
N_Cen_Eu = len(Cen_Eu)
media_Cen_Eu = np.mean(Cen_Eu)
std_Cen_Eu = np.std(Cen_Eu)


Contraste_Cen = contrate_2medias(N_Cen_Lat,N_Cen_Eu,media_Cen_Lat,media_Cen_Eu,std_Cen_Lat,std_Cen_Eu,0.05,alternativa='1=2')

print("\nComo resultado se obtienen los siguientes valores:\n")
print("     Z0                                                         región crítica                                                           p-value\n")
print(Contraste_Cen)





Como resultado se obtienen los siguientes valores:

     Z0                                                         región crítica                                                           p-value

(5.367953557095783, [Interval(-inf, -1.968821974103945, closed='neither'), Interval(1.9688219741039445, inf, closed='neither')], 1.7212752112527596e-07)


b)

Como se puede ver en lo obtenido $Z_0$, se encuentra en la zona critica, por lo que tenemos suficiente evidencia para rechazar $H_0$. Una vez analizado lo anterior se puede saber que la hipotesis que toma validez es la que tiene nuestro interes la cual es que los discursos realizados por presidentes de centro tienen un populismo distinto en las dos regiones.

Por otro lado se tiene el p-value, el cual es el valor minimo del error 1 para el cual poder rechazar $H_0$ y de esa manera aceptar la hipotesis alternativa, si el error 1 es menor al p-value, entonces el rechazo de $H_0$ no se podrá hacer, en caso de que el error 1 sea igual al p_value se va a poder visualizar que para los intervalos de rechazo se tiene lo siguiente [(-inf , $Z_0$),($Z_0$ , inf)]. En este caso se puede observar que el p-value es muy pequeño, por lo que se puede concluir que es poco probable que se rechace $H_0$ incorrectamente.


In [ ]:
#Parte C

Der_Lat = muestra[(muestra['region'] == 'Latin America & Caribbean')| (muestra['lr'] == 1)]
Der_Lat = Der_Lat[Der_Lat['rubricgrade'].notnull()]
Der_Lat = Der_Lat['rubricgrade']

Der_Eu = muestra[(muestra['region'] == 'Western Europe') | (data['lr'] == 1)]
Der_Eu = Der_Eu[Der_Eu['rubricgrade'].notnull()]
Der_Eu = Der_Eu['rubricgrade']

#Parametros Latino
N_Der_Lat = len(Der_Lat)
media_Der_Lat = np.mean(Der_Lat)
std_Der_Lat = np.std(Der_Lat)
#Parametros Europa
N_Der_Eu = len(Der_Eu)
media_Der_Eu = np.mean(Der_Eu)
std_Der_Eu = np.std(Der_Eu)

Contraste_Der = contrate_2medias(N_Der_Lat,N_Der_Eu,media_Der_Lat,media_Der_Eu,std_Der_Lat,std_Der_Eu,0.05,alternativa='1=2')

print("\nComo resultado se obtienen los siguientes valores:\n")
print("     Z0                                                         región crítica                                                           p-value\n")
print(Contraste_Der)



Como resultado se obtienen los siguientes valores:

     Z0                                                         región crítica                                                           p-value

(4.414566642893978, [Interval(-inf, -1.96742838690215, closed='neither'), Interval(1.9674283869021496, inf, closed='neither')], 1.3870836741425308e-05)


c) 

Como se puede ver en lo obtenido $Z_0$, se encuentra en la zona critica, por lo que tenemos suficiente evidencia para rechazar $H_0$. Una vez que se pueda ver lo anterior, se sabe que la hipotesis que toma validez es la que tiene nuestro interés la cual es que los discursos realizados por presidentes de derecha tienen un populismo distinto en las dos regiones.

Al respecto del p-value, al igual que los discursos de los presidentes de centro, se puede observar que este valor es muy pequeño, por lo que se concluye que estadisticamente es muy poco probable que se rechace $H_0$ de manera incorrecta

#Pregunta 2

Considerando solo los discursos realizados en Chile desde el año 2000 en adelante. Diseñe un contraste que permita evaluar la hipótesis de que "*los discursos realizados por presidentes de izquierda (`lr = -1`) son **más populistas** (utilice `rubricgrade`) que los discursos realizados por presidentes de derecha (`lr = -1`)*". Diseñe un contraste de manera que la probabilidad de aceptar incorrectamente esta conjetura sea menor al 5%.  **(10 pts)**

Supongamos por hipótesis nula $H_{o}$ que _Los discursos realizados por presidentes de izquierda son __igual__ de populistas que los discursos realizados por presidentes de derecha_. Así, la hipótesis alternativa $H_{1}$ será que _Los discursos realizados por presidentes de izquierda son __más__ de populistas que los discursos realizados por presidentes de derecha_. Es decir, considerando el contraste mediante hipótesis unilateal y comparando por diferencia de medias con varianza desconocida, esto será: <br>
$\hspace{9cm} H_o :\hspace{2cm}\mu_{izq} = \mu_{der}$<br>$\hspace{9cm} H_1 :\hspace{2cm}\mu_{izq} > \mu_{der}$<br><br>
Abordemos el problema para que el error tipo 1 sea del 5%: 

In [ ]:
error_t1 = 0.05 #alpha

#Muestra considerada eliminando datos desconocidos
muestra = data[data["country"]=="Chile"][data["yearbegin"]>=2000][data["president"]==1][(data["lr"]==-1)|(data["lr"]==1)].dropna()
muestra_p_izq = muestra[muestra["lr"]==-1]
muestra_p_der = muestra[muestra["lr"]==1]

#Media
media_izq = muestra_p_izq["rubricgrade"].mean()
media_der = muestra_p_der["rubricgrade"].mean()

#Desviación estandar
desv_izq = muestra_p_izq["rubricgrade"].std()
desv_der = muestra_p_der["rubricgrade"].std()

#Cantidad de datos
n_izq = len(muestra_p_izq)
n_der = len(muestra_p_der)

#Presentación de datos
Contraste = contrate_2medias(n_izq,n_der,media_izq,media_der,desv_izq,desv_der,error_t1,'1>2') 
print("\nObtenemos los siguientes valores:\n")
print("     Z0                           región crítica                                    p-value\n")
print(Contraste)



Obtenemos los siguientes valores:

     Z0                           región crítica                                    p-value

(0.44547192501661115, [Interval(1.6882977108049164, inf, closed='neither')], 0.32932309676164273)


Dado entonces a que el estadístico es $T_{o}$= 0.445472 no tenemos suficiente evidencia para rechazar la hipótesis nula, pues notemos que el valor límite del intervalo crítico: $t_{n_{X}+n_{Y}-2}^{\alpha}$ = 1.688298 es superior al estadístico obtenido.

Ahora bien, veamos que el p-value será:<br>
$\hspace{10cm} P = P(T>0.445) = 0.33$<br><br>
Es decir, la probabilidad de que el estadístico sea mayor a $T_{o}$ es de un $33$%. No es evidencia fuerte para rechazar la hipótesis. 


#Pregunta 3

En Chile, Michelle Bachelet y Sebastián Piñera han sido dos veces presidente de la república. Por esto, podría resultar interesante determinar si sus niveles de populismo (utilice `rubricgrade`) en sus discursos cambiaron en algo durante sus dos mandatos. 

> a) Ejecute un contraste donde $H_0$ corresponda a que "*los discursos de "Bachelet 1" son **igual o más populistas** que los discursos de "Bachelet 2"* ". Considere $α =  5\%$. **(15 pts)**

> b) Ejecute un contraste donde $H_0$ corresponda a que "*los discursos de "Piñera 1" son **igual o más populistas** que los discursos de "Piñera 2"* ". Considere $α =  5\%$. **(15 pts)**

a) Para esta pregunta, tendremos que nuestra hipótesis es la siguiente:

+ $ H_0 $ = Los discursos de Bachelet 1 son igual o más populistas que el de Bachelet 2

+ $ H_1 $ = Los discursos de Bachelet 1 son menos populistas que el de Bachelet 2

In [ ]:
# Lee los datos desde el dataset
data = pd.read_csv("gpd_v2_20220427.csv")

# Filtra por el presidente 'Michele Bachelet'
data_bachelet = data[data['leader'] == 'Michelle Bachelet']
data_bachelet = data_bachelet[['leader', 'yearbegin', 'rubricgrade']]
data_bachelet.dropna()

# Filtra por el índice de populismo (0-2)
data_bachelet_populism = data_bachelet.rubricgrade

Para suplir nuestra falta de conocimiento general y para más certeza, usamos Python para saber en que años la presidenta asumió el mandato, para así poder corroborar que tuvo dos periodos presidenciales y para saber como filtrar los datos.

In [ ]:
column_list = data_bachelet.columns.values.tolist()
print(data_bachelet['yearbegin'].unique())

[2014 2006]


Ahora usamos un contraste de 2 medias con alternativa bilateral, usando el $\alpha = 0.05$

In [ ]:
# Filtramos los datos por las fechas de mandato
data_bachelet_2014 = data_bachelet[data_bachelet['yearbegin'] == 2014]
data_bachelet_2006 = data_bachelet[data_bachelet['yearbegin'] == 2006]
data_bachelet_2014_populism = data_bachelet_2014['rubricgrade']
data_bachelet_2006_populism = data_bachelet_2006['rubricgrade']

# Obtenemos la poblacion para cada grupo
print("El n de el primer y segundo periodo presidencial de Bachelet")
n1 = len(data_bachelet_2006_populism)
n2 = len(data_bachelet_2014_populism)
print("   n1 =", n1)
print("   n2 =", n2)

# Obtenemos la media para cada grupo
print("La media entre el primer y segundo periodo presidencial de Bachelet")
media1 = data_bachelet_2006_populism.mean()
media2 = data_bachelet_2014_populism.mean()
print("   media1 =", media1)
print("   media2 =", media2)

# Obtenemos la desviación estándar
print("La desviación estándar entre el primer y segundo periodo presidencial de Bachelet")
std1 = np.std(data_bachelet_2006_populism)
std2 = np.std(data_bachelet_2014_populism)
print("   std1 =", std1)
print("   std2 =", std2)

# Mostramos el alpha pedido
print("Nuestro alpha")
error_tipo1 = 0.05
print("   a =", error_tipo1)

# Calculamos el contraste
print("Contraste de H0")

print("\nObtenemos los siguientes valores:\n")
print("     Z0                           región crítica                                    p-value\n")
contraste_h0 = contrate_2medias(n1,n2,media1,media2,std1,std2,error_tipo1,alternativa='1>2')
print("   ", contraste_h0)

El n de el primer y segundo periodo presidencial de Bachelet
   n1 = 16
   n2 = 16
La media entre el primer y segundo periodo presidencial de Bachelet
   media1 = 0.0
   media2 = 0.2333333333333333
La desviación estándar entre el primer y segundo periodo presidencial de Bachelet
   std1 = 0.0
   std2 = 0.1885618083164127
Nuestro alpha
   a = 0.05
Contraste de H0

Obtenemos los siguientes valores:

     Z0                           región crítica                                    p-value

    (-4.949747468305832, [Interval(1.7530503556925547, inf, closed='neither')], 0.9999127035584517)


Nuestra hipótesis inicial $H_0$ era que los discursos de la presidenta eran más populistas en su primer periodo en relación al segundo. Podemos ver que en el contraste que realizamos, que nuestro $Z_0$ = $-4.94$ está fuera de la *región crítica* = $[1.753,∞]$, lo cuál puede ser un indicio de que la hipótesis es correcta. Luego, vemos el *p-value*, el cuál representa el error mínimo que debe tomar $\alpha$ para poder rechazar, en este caso es evidente que está muy alejado del $\alpha$ que usamos. Como no tenemos argumentos para rechazar, podemos decir que nuestra Hipótesis puede ser correcta.

b) Para esta pregunta, repetimos el mísmo procesoq que en el punto anterior:

+ $ H_0 $ = Los discursos de Piñera 1 son igual o más populistas que el de Piñera 2

+ $ H_1 $ = Los discursos de Piñera 1 son menos populistas que el de Piñera 2

In [ ]:
# Filtra por el presidente 'Sebastián Piñera'
data_pinera = data[data['leader'] == 'Sebastian Pinera']
data_pinera = data_pinera[['leader', 'yearbegin', 'rubricgrade']]
data_pinera.dropna()

# Filtra por el índice de populismo (0-2)
data_pinera_populism = data_pinera.rubricgrade

# Busca los distintos periodos de Piñera
column_list = data_pinera.columns.values.tolist()
print("Los diferentes periodos de Piñera")
print("   T =", data_pinera['yearbegin'].unique())

Los diferentes periodos de Piñera
   T = [2018 2010]


In [ ]:
# Filtramos los datos por las fechas de mandato
data_pinera_2010 = data_pinera[data_pinera['yearbegin'] == 2010]
data_pinera_2018 = data_pinera[data_pinera['yearbegin'] == 2018]
data_pinera_2010_populism = data_pinera_2010['rubricgrade']
data_pinera_2018_populism = data_pinera_2018['rubricgrade']
data_pinera_2010.dropna()
data_pinera_2018.dropna()

# Obtenemos la poblacion para cada grupo
print("El n de el primer y segundo periodo presidencial de Piñera")
n1 = len(data_pinera_2010)
n2 = len(data_pinera_2018)
print("   n1 =", n1)
print("   n2 =", n2)

# Obtenemos la media para cada grupo
print("La media entre el primer y segundo periodo presidencial de Piñera")
media1 = data_pinera_2010_populism.mean()
media2 = data_pinera_2018_populism.mean()
print("   media1 =", media1)
print("   media2 =", media2)

# Obtenemos la desviación estándar
print("La desviación estándar entre el primer y segundo periodo presidencial de Piñera")
std1 = np.std(data_pinera_2010_populism)
std2 = np.std(data_pinera_2018_populism)
print("   std1 =", std1)
print("   std2 =", std2)

# Mostramos el alpha pedido
print("Nuestro alpha")
error_tipo1 = 0.05
print("   a =", error_tipo1)

# Calculamos el contraste
print("Contraste de H0")

print("\nObtenemos los siguientes valores:\n")
print("     Z0                           región crítica                                    p-value\n")
contraste_h0 = contrate_2medias(n1,n2,media1,media2,std1,std2,error_tipo1,alternativa='1>2')
print("   ", contraste_h0)

El n de el primer y segundo periodo presidencial de Piñera
   n1 = 16
   n2 = 16
La media entre el primer y segundo periodo presidencial de Piñera
   media1 = 0.0
   media2 = 0.15
La desviación estándar entre el primer y segundo periodo presidencial de Piñera
   std1 = 0.0
   std2 = 0.206155281280883
Nuestro alpha
   a = 0.05
Contraste de H0

Obtenemos los siguientes valores:

     Z0                           región crítica                                    p-value

    (-2.910427500435996, [Interval(1.7530503556925547, inf, closed='neither')], 0.9946177121196285)


Nuestra hipótesis inicial $H_0$ era que los discursos del presidente eran más populistas en su primer periodo en relación al segundo. Podemos ver que en el contraste que realizamos, que nuestro $Z_0$ = $-4.94$ está fuera de la *región crítica* = $[1.753,∞]$, lo cuál puede ser un indicio de que la hipótesis es correcta. Luego, vemos el *p-value*, el cuál representa el error mínimo que debe tomar $\alpha$ para poder rechazar, en este caso es evidente que está muy alejado del $\alpha$ que usamos. Como no tenemos argumentos para rechazar, podemos decir que nuestra Hipótesis puede ser correcta.

#Pregunta 4

En los últimos 20 años Argentina ha aumentado su nivel de pobreza, donde más del 25% de la población del país se encuentra en esta situación.

> a) Diseñe un contraste que permita evaluar la hipótesis de que "*el populismo (utilice `rubricgrade`) promedio en los discursos de presidentes argentinos es mayor a 0.5*". Diseñe un contraste de manera que la probabilidad de aceptar incorrectamente esta conjetura sea menor al 5%. **(10 pts)**

> b) Diseñe un contraste que permita evaluar la hipótesis de que "*los discursos realizados por presidentes argentinos son más populistas que los discursos realizados por el resto de países de `region=Latin America & Caribbean`*". Diseñe un contraste de manera que la probabilidad de aceptar incorrectamente esta conjetura sea menor al 5%  **(10 pts)**

Como supuesto se va a tomar los discursos hechos entre 2000 y 2022, para las dos preguntas, esta aclaración se hace por la contextualización de la pregunta. 
Ademas se tiene que $H_0$ sea que los discursos realizados por presidentes argentinos son igual o menor a 0.5 y el caribe. Por lo que $H_1$ es igual a que los discursos argentinos son mas populistas que 0.5

a) 

El contraste evaluado para evaluar la hipotesis que se utilizará, va a ser el Test de una media con varianza desconocida. Con lo que se obtiene:

$H_0$: popArg $\leqslant$ 0.5

$H_1$: popArg $>$ 0.5

In [ ]:
Arg = data[data['country'] == 'Argentina']
Arg = Arg[Arg['rubricgrade'].notnull()]
Arg = Arg['rubricgrade']

N_Arg = len(Arg)
std_Arg = np.std(Arg)
media_Arg = np.mean(Arg)

Con_Arg = contrate_1media(N_Arg,media_Arg,std_Arg,0.5,0.05,alternativa='1>0')


print("\nComo resultado se obtienen los siguientes valores:\n")
print("     Z0                                 región crítica                              p-value\n")
print(Con_Arg)




Como resultado se obtienen los siguientes valores:

     Z0                                 región crítica                              p-value

(-0.48603649954805606, [Interval(1.669013024840352, inf, closed='neither')], 0.6856993326586565)


Con los resultados obtenidos, no se tiene la evidencia suficiente para poder rechazar $H_0$, esto se debe a que el estadistico no esta dentro de la región crítica, por lo que se puede decir que el populismo promedio en los discursos de presidentes argentinos es menor o igual a 0.5 entre los años del 2000 al 2022

b)

Tomemos como supuesto que $H_0$ sea que los discursos realizados por presidentes argentinos son igual o menos populistas que los discursos realizados por el resto de países de la region Latino America y el caribe. Por lo que $H_1$ es igual a que los discursos argentinos son mas populistas que en los otros paises de la region Latino America y el caribe.

Para este caso el contraste utilizado va a ser el Test de dos medias con varianzas desconocidas

$H_0$: $\mu$popArg $\leqslant$ $\mu$popLat

$H_1$: $\mu$popArg $>$ $\mu$popLat

In [ ]:
Lat = data[data['region'] == 'Latin America & Caribbean']
Lat = Lat[Lat['country'] != 'Argentina']
Lat = Lat[Lat['rubricgrade'].notnull()]
Lat = Lat['rubricgrade']

N_Lat = len(Lat)
std_Lat = np.std(Lat)
media_Lat = np.mean(Lat)

Con_Diff = contrate_2medias(N_Arg,N_Lat,media_Arg,media_Lat,std_Arg,std_Lat,0.05,alternativa='1>2')


print("\nComo resultado se obtienen los siguientes valores:\n")
print("     Z0                                 región crítica                              p-value\n")
print(Con_Diff)


Como resultado se obtienen los siguientes valores:

     Z0                                 región crítica                              p-value

(-0.27015369791507815, [Interval(1.6648845371855607, inf, closed='neither')], 0.6061176182944319)


Dado que el estadistico de contraste no esta dentro de la región crítica, no hay evidencia suficiente para poder rechazar $H_0$, por lo que se podría decir que los discursos realizados por presidentes argentinos son menos o igual de populistas que los discursos realizados por el resto de países de la region Latino America y el Caribe entre los años del 2000 al 2022

# Disclaimer 

Los datos aquí utilizados fueron obtenidos usando la metodología detallada en [1] y por lo tanto no necesariamente reflejan la opinión de los profesores o ayudantes de este curso. Cualquier alusión a personas o países es meramente circustancial y tiene sólo fines pedagógicos. Todo público puede acceder a los datos en [2] 

[1] Hawkins, K. A., Aguilar, R., Silva, B. C., Jenne, E. K., Kocijan, B., & Kaltwasser, C. R. (2019, June). Measuring populist discourse: The global populism database. In EPSA Annual Conference in Belfast, UK, June (pp. 20-22).

[2] Hawkins, Kirk A.; Aguilar, Rosario; Castanho Silva, Bruno; Jenne, Erin K.; Kocijan, Bojana; Rovira Kaltwasser, Cristóbal, 2019, "Global Populism Database", https://doi.org/10.7910/DVN/LFTQEZ, Harvard Dataverse, V2

# Código de Honor UTFSM
La realización de este taller de adhiere al código de honor de la Universidad Federico Santa María.